docs/integration/DataTypingAndSchemasHandling.md

```diff
@@ -0,0 +1,246 @@
+| [enum](https://json-schema.org/understanding-json-schema/reference/generic.html#enumerated-values) | String     |                                                                                                                                                                    |
+| [array](https://json-schema.org/understanding-json-schema/reference/array.html)                    | List[_]    |                                                                                                                                                                    |
+
+#### [String Format](https://json-schema.org/understanding-json-schema/reference/string.html#format)
+
+| JSON Schema | Java type     | Sample                    | Comment                |
+|-------------|---------------|---------------------------|------------------------|
+| date-time   | ZonedDateTime | 2021-05-17T07:34:00+01:00 |                        |
+| date        | LocalDate     | 2021-05-17                | Timezone is not stored |
+| time        | LocalTime     | 07:34:00.12345            | Timezone is not stored |
+
+#### [Objects](https://json-schema.org/understanding-json-schema/reference/object.html)
+
+| object configuration                                                      | Java type            | Comment                                          |
+|---------------------------------------------------------------------------|----------------------|--------------------------------------------------|
+| object with properties                                                    | Map[String, _]       |                                                  |
+| object with properties and enabled `additionalProperties`                 | Map[String, _]       | Additional properties are available at  runtime. |
+| object without properties and `additionalProperties: true`                | Map[String, Unknown] | It's similar to avro map.                        |
+| object without properties and `additionalProperties: {"type": "integer"}` | Map[String, Integer] | It's similar to avro map.                        |
+
+We support `additionalProperties`, but additional fields won't be available in the hints on the Designer. To get
+an additional field you have to do `#inpute.get("additional-field")`, but remember that result of this expression is
+depends on `additionalProperties` type configuration and can be `Unknown`.
+
+#### [Schema Composition](https://json-schema.org/understanding-json-schema/reference/combining.html)
+
+| type  | Java type                                  | Comment                       |
+|-------|--------------------------------------------|-------------------------------|
+| oneOf | Any from the available list of the schemas | We treat it just like a union |
+| anyOf | Any from the available list of the schemas | We treat it just like a union |
```

Comment:
`We treat it just like a union` - some example would be helpful...
What's more, currently we don't support extending scheme with composition, e.g.:
```
{
   "type":"object",
 	"properties": {
        "f3": {"type": "string"}
     },
     "anyOf": [
          { "type": "object", "properties": {"f1": {"type": "string"}}, "additionalProperties": false, "required": ["f1"] },
          { "type": "object", "properties": {"f2": {"type": "string"}}, "additionalProperties": false, "required": ["f2"]  },
        ],
       "required": ["f3"]
}
```

Reply:
Belove is information about composition:
```
| [composition](https://json-schema.org/understanding-json-schema/reference/combining.html) | [allOf](https://json-schema.org/understanding-json-schema/reference/combining.html#allof), [not](https://json-schema.org/understanding-json-schema/reference/combining.html#not)                                                                                                                                                                                                                                                                                                                                                                                                                                           | Read more about [validation modes](#validation-and-encoding). |
```